applications/RANSApplication/custom_utilities/rans_variable_utilities.cpp

```diff
@@ -147,33 +104,11 @@ double GetMaximumScalarValue(
     const auto& r_communicator = rModelPart.GetCommunicator();
     const auto& r_nodes = r_communicator.LocalMesh().Nodes();
 
-    class MaxReducer{
-        public:
-            typedef double value_type;
-            double max_value = std::numeric_limits<double>::lowest();
-
-            value_type GetValue()
-            {
-                return max_value;
-            }
-
-            void LocalReduce(const double max_value){
-                this->max_value = std::max(this->max_value, max_value);
-            }
-            void ThreadSafeReduce(MaxReducer& rOther){
-                #pragma omp critical
-                {
-                    this->max_value = std::max(this->max_value, rOther.max_value);
-                }
-            }
-    };
-
     const int number_of_nodes = r_nodes.size();
     const double max_value =
-        IndexPartition<int>(number_of_nodes).for_each<MaxReducer>(
-            [&](const int i) -> double {
-                return (r_nodes.begin() + i)->FastGetSolutionStepValue(rVariable);
-            });
+        IndexPartition<int>(number_of_nodes).for_each<MaxReduction<double>>([&](const int i) -> double {
+            return (r_nodes.begin() + i)->FastGetSolutionStepValue(rVariable);
+        });
```

Comment:
I think you can use the `block_for_each` in here.

Reply:
Unfortunately not, again the problem of `const ModelPart`